release-19.2: Install `go-test-teamcity`

rickystewart opened the pull request:
This binary was uninstalled from our build agents because it was tought
to be unused, but 19.2 builds still need it. Manually install it in this
branch.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/58719)
<!-- Reviewable:end -->

rickystewart committed:
release-19.2: Install `go-test-teamcity`

This binary was uninstalled from our build agents because it was tought
to be unused, but 19.2 builds still need it. Manually install it in this
branch.

Release note: None

knz commented:
The issue with linting is likely that `crlfmt` was not vendored properly. The version of `crlfmt` to use in linting should be the one in the 19.2 branch, or a SHA in the crlfmt repo that was valid when 19.2 was released.

rickystewart commented:
The only differences between the most recent version of `crlfmt` pre-19.2 (`3c56c65`) and what's currently vendored are a few test files. To be clear, you're saying that linting will succeed if I copy these over to `vendored`? Just clarifying because I don't see any other way to interpret your comment, but I also don't see how that would fix the issue.

```
crlfmt$ diff -r . ../cockroach/vendor/github.com/cockroachdb/crlfmt/
Only in .: .git
Only in .: main_test.go
Only in .: testdata
crlfmt$
```

bdarnell commented:
Then it sounds like it's vendored correctly, but it may be running something other than the vendored version? 

Anyway, I would support skipping the crlfmt lint check in release-19.2 if the problem turns out to be non-trivial.

knz commented:
I may have been mistaken. What you propose is not what I had intended to mean.

My reading of the test failure suggests that the `crlfmt` currently running in CI for 19.2 is not the same that we were using when 19.2 was released.

I don't know why that is the case for sure. my hypothesis is that the crlfmt version is not the same because of a vendoring fail, but maybe that hypothesis is wrong.

It's also possible that we have a copy of crlfmt in vendor but the one installed on the agent and used by the linter *is another one* (we had that problem for other tools in the past)

bdarnell commented:
But I'm not sure the problem is crlfmt-related. I think it's incorrect to check in changes to Gopkg.toml without the corresponding changes to Gopkg.lock. You need to run `bin/dep ensure` after changing the toml file and then it will regenerate the lock file.

knz commented:
There are two sets of problems in the lint failure

- the gopkg locks, which I assumed were well-understood already. The proper `dep` command must be run and the resulting changes must be checked in. I did not bother to comment on that in my previous review, but yes, that needs to be fixed.

- separately there's a bunch of crlfmt-generated diffs. that's what I was looking at.

rickystewart commented:
OK, thanks for the tip about `bin/dep ensure`, that's what I was missing. I updated that. If the problem turns out to be `crlfmt`-related then that will require some other work, not sure about that yet.

rickystewart commented:
Fixed `timestamp.pb.go`.

bdarnell reviewed (approved):
LGTM as long as we get a green build this time.

bdarnell commented:
This failure looks like an unpinned dependency in a GSSAPI integration test. I would just ignore that failure and merge anyway so we can proceed.
